src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -2770,6 +2862,18 @@ public static decimal Average<TSource>(this IEnumerable<TSource> source, Func<TS
         }
     }
 
+    /// <summary>
+    /// An iterator that can (or sometimes can) produce an array through an optimised path.
+    /// </summary>
+    internal interface IArrayProvider<TElement>
+    {
+        /// <summary>
+        /// Produce an array of the sequence through an optimised path, if possible.
+        /// </summary>
+        /// <returns>The array, or null if an optimised path isn't possible, and default behaviour should be used.</returns>
```

Comment:
I realize this is very US of me, but would you mind changing "optimised" to "optimized"?